src/cargo/ops/tree/mod.rs

```diff
@@ -179,10 +179,17 @@ pub fn build_and_print(ws: &Workspace<'_>, opts: &TreeOptions) -> CargoResult<()
         .map(|pkg| (pkg.package_id(), pkg))
         .collect();
 
+    // TODO build multiple graphs, one for each resolved features set.
+    let resolved_features = &ws_resolve
```

Comment:
> cargo tree - I just use the first feature set. This is definitely not ideal, but I'm not entirely sure what's the correct solution here. Printing multiple trees? Disallowing this, forcing users to select only one package?

I see this as no different than `cargo tree --workspace` where there will be multiple root packages.  `--edges features` (ro whatever its called) will be needed to fully understand it.

Reply:
True, I got confused, this part is fine. I think that just signalling to the user whether features are unified between the trees or not is a new problem.